### PR TITLE
feat(ui-toolkit,governance): description preview and read more pattern

### LINKED ISSUE #4091: Description preview and read more pattern
## Story

As a user
I want to see enough of a proposal description so I understand what this proposal is about without needing to expand all the collapsed sections
So that I can quickly work out if I want to learn more about this proposal or not

## Acceptance Criteria

- [ ] I can see the first few lines of the summary description when I first land on a proposal [ rationale / description ]  
- [ ] Vertical height should be fairly constant across diff screen widths
- [ ] There is a way to read the full description (Show More button) 
- [ ] There is a way to "collapse" back e.g. "see less"

## Background
e.g. read more pattern Image

https://snapshot.org/#/stgdao.eth/proposal/0x108a9e597560c4f249cd8be23acd409059fcd17bb2290d69a550ac2232676e7d

## Tasks

- [x] UX (if needed)
- [x] Design (if needed)
- [ ] Team and stakeholder review
- [ ] Specs reviewed and created or adjusted
- [ ] Implementation
- [ ] Testing (unit and/or e2e)
- [ ] Code review
- [ ] QA review

## Design

[See the design solution in situ in Figma here](https://www.figma.com/file/X9l0t8TZUF1tU8u0ILV3ws/Governance?type=design&node-id=2846%3A23539&mode=design&t=oMTYn2eS4bYPDQGK-1) ↳ 

Screenshot 2023-06-27 at 12 26 13

### PR DESCRIPTION
**Pull request by sam-keen**
# Related issues 🔗

Closes #4091

# Description ℹ️

Adds a 'show more' component to the ui-toolkit, which expands automatically if the content height is less than the set closed max height (i.e. it's responsive to changes in width that affect content height). Also accepts custom closed max height and custom overlay colour to work alongside backgrounds that aren't white or black. 

Also makes use of the component in the proposal description section of proposal details on Governance.

# Demo 📺

Screenshot 2023-08-22 at 15 37 51

### CODE REVIEW

File: libs/ui-toolkit/src/components/show-more/show-more.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      overlayColourOverrides ? overlayColourOverrides : 'to-white dark:to-black'
+    }`,
+    {
+      hidden: expanded,
+    }
+  );
+
+  return (
+    <>
+      <div className="relative">
```
Comment (edd) [inline libs/ui-toolkit/src/components/show-more/show-more.tsx:58]: **nitpick (non-blocking)**: My only comment is that this seems quite 'div' heavy, but... it really doesn't matter

File: libs/ui-toolkit/src/components/show-more/show-more.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          ref={containerRef}
+          className={containerClasses}
+          style={{ maxHeight: expanded ? 'none' : `${closedMaxHeightPx}px` }}
+        >
+          {children}
+        </div>
+        <div className={overlayClasses}></div>
+      </div>
+
+      {!expanded && (
```
Comment (edd) [inline libs/ui-toolkit/src/components/show-more/show-more.tsx:69]: **thought (non-blocking)**: I hadn't read the ticket for this so the logic surprised me. I'm. fine with there being no 'collapse' button, but I wonder if we'll get feedback asking for one? Probably not. Perhaps we will from other apps that use this component. Easy enoguh to add when it's relevant.

Screenshot 2023-08-22 at 16 39 10
Screenshot 2023-08-22 at 16 39 15

**Review by edd — APPROVED** [review]

File: libs/ui-toolkit/src/components/show-more/show-more.tsx (hunk 2 above)
Comment (sam-keen) [reply libs/ui-toolkit/src/components/show-more/show-more.tsx:69]: I ummed and ahhed about it. In the end, I decided to keep it simple, but you make a good point, I'll think some more.

Comment (sam-keen) [reply libs/ui-toolkit/src/components/show-more/show-more.tsx:69]: Based on our conversation, I'll keep it simple for now

**Pull request merged**